### PR TITLE
Implemented split pane and used in new markdown editor.

### PR DESCRIPTION
**Pull request by MattPutnam**
We'll need a split pane for the IGV stuff, according to Alex's mockup.  Proof of concept is a side-by-side markdown editor/preview.

This split pane implementation is probably incomplete; the scrolling/overflow behavior has
been tailored to work specifically in this case.  It will likely need to expose style
hooks to be used with other types of child components.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: Tell tech lead that the PR exists if s/he wants to look at it
- [x] **Submitter**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **Tech lead**: sign off
- [x] **LR**: sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't 
  working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email

### CODE REVIEW

**Comment by MatthewBemis** [comment]
This looks pretty awesome. I couldn't get it to work in top/bottom configuration, though.

I changed the SplitPlane props to: 

```

[comps/SplitPane
  {:top text-area
   :bottom markdown-view
...
```

and the preview didn't work

**Comment by MattPutnam** [comment]
Yeah, right now you have to style the child components to behave correctly.  Ideally you wouldn't have to do that but CSS overflow stuff is fiddly.

File: src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs [hunk 1]
```diff
@@ -346,3 +346,48 @@
...
+   (fn [{:keys [props]}]
+     {:slider-position (or (:initial-slider-position props) 100)})
+   :render
+   (fn [{:keys [props state]}]
+     (let [{:keys [left right top bottom]} props]
+       (assert (or (and left right) (and top bottom)) "Either specify left/right or top/bottom for SplitPane")
+       [:div {:style {:display "flex" :flexDirection (if left "row" "column")}}
+        [:div {:style {:flexGrow 0 :flexShrink 0 :flexBasis (:slider-position @state)}}
+         (or left top)]
+        [:div {:style {:flex "0 0 10px"
```
Comment (davidangb) [inline src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs:363]: ui tweak: could you put a "grabby" indicator in this divider? Something like the font-awesome bars: http://fontawesome.io/icon/bars/, rotated as appropriate for left/right vs. top/bottom.

**Comment by davidangb** [comment]
minor UI comment inline. Related to @MatthewBemis 's comment, if top/bottom isn't fully implemented, change the assert to only allow left/right for now. Beyond these comments, 👍 as TL.

**Comment by MattPutnam** [comment]
Top/bottom works, it just plays really poorly with these two components.

**Comment by MatthewBemis** [comment]
:+1:

**Pull request merged**